deploy/main.tf

```diff
@@ -43,8 +43,21 @@ chmod 0600 /root/nixbuild.pem
 EOF
 }
 
-output "server_ip_addr" {
-  value = aws_instance.binplz_server.public_ip
+output "public_ip_addr" {
+  value = aws_eip.binplz_eip.public_ip
+}
+
+resource "aws_eip" "binplz_eip" {
```

Comment:
I didn't realize that elastic IPs don't actually cost anything as long as they are associated with an instance and the instance is running:

https://aws.amazon.com/premiumsupport/knowledge-center/elastic-ip-charges/